tests/repo_finder/test_provenance_extractor.py

```diff
@@ -0,0 +1,474 @@
+
+
+def test_witness_gitlab_is_valid(witness_gitlab_provenance: dict[str, JsonType]) -> None:
+    """Test valid Witness v0.1 GitLab provenance."""
+    _perform_provenance_comparison(
+        witness_gitlab_provenance,
+        "https://gitlab.com/tinyMediaManager/tinyMediaManager",
+        "cf6080a92d1c748ba5f05ea16529e05e5c641a49",
+    )
+
+
+def test_witness_github_is_valid(
+    witness_github_provenance: dict[str, JsonType], target_repository: str, target_commit: str
+) -> None:
+    """Test valid Witness v0.1 GitHub provenance."""
+    _perform_provenance_comparison(witness_github_provenance, target_repository, target_commit)
+
+
+@pytest.mark.parametrize(
+    ("keys", "new_value", "attestation_index"),
+    [
+        (["attestation", "projecturl"], "", 0),
+        (["attestation", "projecturl"], None, 0),
+        (["attestation", "commithash"], "", 1),
+        (["attestation", "commithash"], None, 1),
+    ],
+)
+def test_witness_github_is_invalid(
+    witness_github_provenance: dict[str, JsonType], keys: list[str], new_value: JsonType, attestation_index: int
+) -> None:
+    """Test invalidly modified Witness v0.1 GitHub provenance."""
+    attestations = json_extract(witness_github_provenance, ["predicate", "attestations"], list)
+    _json_modify(attestations[attestation_index], keys, new_value)
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(witness_github_provenance, "", "")
+
+
+def test_witness_github_remove_attestation(witness_github_provenance: dict[str, JsonType]) -> None:
+    """Test removing Git attestation from Witness V0.1 GitHub provenance."""
+    attestations = json_extract(witness_github_provenance, ["predicate", "attestations"], list)
+    _json_modify(witness_github_provenance, ["predicate", "attestations"], attestations[:1])
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(witness_github_provenance, "", "")
+
+
+@pytest.mark.parametrize(
+    ("type_", "predicate_type"),
+    [
+        ("https://in-toto.io/Statement/v0.1", "https://slsa.dev/provenance/v1"),
+        ("https://in-toto.io/Statement/v1", "https://slsa.dev/provenance/v0.2"),
+        ("https://in-toto.io/Statement/v1", "https://slsa.dev/provenance/v0.1"),
+        ("https://in-toto.io/Statement/v1", "https://witness.testifysec.com/attestation-collection/v0.1"),
+    ],
+)
+def test_invalid_type_payloads(type_: str, predicate_type: str) -> None:
+    """Test payloads with invalid type combinations."""
+    payload_text = '{ "_type": ' + f'"{type_}",' + ' "predicateType": ' + f'"{predicate_type}",'
+    payload_text = f"{payload_text}" + '"subject": [], "predicate": {} }'
+    payload = json.loads(payload_text)
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+
+def _perform_provenance_comparison(payload: dict[str, JsonType], expected_repo: str, expected_commit: str) -> None:
```

Comment:
`expected_repo` and `expected_commit` can be made optional parameters so that we don't have to pass in empty strings when no comparison is needed.

Also, this function is literally testing the `extract_repo_and_commit_from_provenance` function. So I think why not call it `_test_extract_repo_and_commit_from_provenance`. It is much more clear than `_perform_provenance_comparison`.

```suggestion
def _test_extract_repo_and_commit_from_provenance(
    payload: dict[str, JsonType],
    expected_repo: str | None = None,
    expected_commit: str | None = None,
) -> None:
```